src/clusterfuzz/_internal/tests/core/swarming/swarming_test.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Swarming tests."""
+import base64
+import unittest
+
+from clusterfuzz._internal import swarming
+from clusterfuzz._internal.datastore import data_types
+from clusterfuzz._internal.protos import swarming_pb2
+from clusterfuzz._internal.tests.test_libs import helpers
+from clusterfuzz._internal.tests.test_libs import test_utils
+
+
+@test_utils.with_cloud_emulators('datastore')
+class GetSpecFromConfigTest(unittest.TestCase):
```

Comment:
There are a couple of untested branches in `_get_new_task_spec()`:

- when there is no matching instance spec, we raise a `ValueError`
- when the command is `"fuzz"`, we use the special fuzzing session duration
- when the docker image is unspecified on the instance spec, we use an empty string instead